role-service/src/test/java/uk/ac/ox/ndph/mts/role_service/controller/PermissionControllerTest.java

```diff
@@ -0,0 +1,50 @@
+package uk.ac.ox.ndph.mts.role_service.controller;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+import uk.ac.ox.ndph.mts.role_service.model.Permission;
+import uk.ac.ox.ndph.mts.role_service.model.PermissionRepository;
+
+import java.util.Collections;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@WebMvcTest(PermissionController.class)
+class PermissionControllerTest {
+
+
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private PermissionRepository permissionRepo;
+
+    @Test
+    void whenGetPaged_thenReceiveSuccess()
+            throws Exception {
+        Permission dummyPermission = new Permission();
+        String dummyId="dummy-permission";
+        dummyPermission.setId(dummyId);
+
+        when(permissionRepo.findAll(PageRequest.of(0, 10))).thenReturn(new PageImpl(Collections.singletonList(dummyPermission)));
+        mvc.perform(get("/permissions?page=0&size=10"))
```

Comment:
In another task I've used MvcUriComponentsBuilder to get the uri of the method so that it won't be listed here and in the tests. What do you think about something like:
`    private String getRoleAssignmentUri() {
        return MvcUriComponentsBuilder.fromMethodCall(on(PractitionerController.class)
                .saveRoleAssignment("987", null)).build().toUriString();
    }`

Reply:
I am not sure.   It adds a lot of complexity and is opaque.   I see the benefit of dynamically discovering the URI, in case it changes.   But is this benefit worth the cost - KISS / self documenting etc?   Not a strong resistance - happy to switch to this if you think it's best - we should standardise one way or the other across all services?

Reply:
We decided to go with variables in the tests to make it tidy &simple

Reply:
We discussed and agreed to use constants for uris in tests instead.  I have updated RoleControllerTest as well as PermissionControllerTest.